rustls/src/crypto/cipher.rs

```diff
@@ -1,11 +1,49 @@
-use crate::error::Error;
+use crate::enums::{ContentType, ProtocolVersion};
+use crate::error::{Error, PeerMisbehaved};
 use crate::msgs::codec;
-use crate::msgs::message::{BorrowedPlainMessage, OpaqueMessage, PlainMessage};
+use crate::msgs::fragmenter::MAX_FRAGMENT_LEN;
+pub use crate::msgs::message::{BorrowedPlainMessage, OpaqueMessage, PlainMessage};
 
 /// Objects with this trait can decrypt TLS messages.
 pub trait MessageDecrypter: Send + Sync {
     /// Perform the decryption over the concerned TLS message.
     fn decrypt(&self, m: OpaqueMessage, seq: u64) -> Result<PlainMessage, Error>;
+
+    /// For TLS1.3 (only), checks the length msg.payload is valid and removes the padding.
+    fn tls13_check_length_and_unpad(&self, mut msg: OpaqueMessage) -> Result<PlainMessage, Error> {
```

Comment:
If we're going to make this public I think we should also expose `OpaqueMessage` and `PlainMessage` as part of the public API (as in, other than via `internals`), as opaque types.

Also, I don't love how this is attached to the `MessageDecrypter` while it doesn't actually use `self` anywhere. If we're going to expose `OpaqueMessage`, maybe this becomes a method on it?